dpypelines/pipeline/dataset_ingress_v1.py

```diff
@@ -100,11 +100,14 @@ def dataset_ingress_v1(files_dir: str, pipeline_config: dict):
 
     # Get Upload Service URL from environment variable
     try:
-        upload_url = os.environ.get("UPLOAD_SERVICE_URL", None)
-        assert (
-            upload_url is not None
-        ), "UPLOAD_SERVICE_URL environment variable not set."
-        logger.info("Got Upload Service URL", data={"upload_url": upload_url})
+        if os.environ["SKIP_DATA_UPLOAD"] is False:
```

Comment:
You need to handle the case where `SKIP_DATA_UPLOAD` is not set as an env var - there's some code on L360-L376 that you could move up (which also converts str to bool). Then do something like:

```
if skip_data_upload is False:
    try:
        upload_url = os.environ.get("UPLOAD_SERVICE_URL", None)
        assert (upload_url is not None), "UPLOAD_SERVICE_URL environment variable not set"
        logger.info("Got Upload Service URL", data={"upload_url": upload_url})
    except Exception as err:
        logger.error("Error occurred when getting Upload Service URL", err)
        de_notifier.failure()
        raise err
```

I think this should also work for `generic_ingress_v1`

Reply:
okay made the changes but added and else statement cause the `upload_url` would not be assigned. This way the solution would require much change

Reply:
made the changes, I did add an else part in cause the `UploadServiceClient` is using the `upload_url`.

Reply:
But if `SKIP_DATA_UPLOAD` isn't False, there won't be anything to upload, so you don't need the else statement because you don't need to set `upload_url`. Also, at the moment, because you're not calling `str_to_bool` on `SKIP_DATA_UPLOAD`, if it's set, then it will always be True (because it's a non-empty string) - you need to move the code from L361-376 to above this try-except block.

Reply:
Sorted :)